File: day20/part2.cpp
#include <iostream>
#include <math.h>
#include <stdlib.h>

using namespace std;

int main () {
	int minimum = 33100000;
	int gifts = 0;
	int house = 1;

	while (gifts < minimum) {
		gifts = 0;

		int squareRoot = (int) sqrt(house) + 1;
		for (int i = 1; i < squareRoot; i++) {
			if (house % i == 0) {
				int pair = house / i;
				if (pair < 50)
					gifts += 11 * i;
				if (pair != i && house / pair < 50)
					gifts += 11 * pair;
			}
		}

		house++;
	}

	cout << house - 1 << endl;
}

File: 2015/ruby/day20/part2.rb
#!/usr/bin/ruby

minimum = 33100000
house = 0
gifts = 0

while gifts * 11 < minimum
    house += 1
    gifts = 0

    for elf in 1..Math.sqrt(house).to_i
        if house % elf == 0
            complement = house / elf

            if elf < 50
                gifts += complement
            end

            if complement < 50 and elf != complement
                gifts += elf
            end
        end
    end
end

puts house
